src/functions/index.ts

```diff
@@ -74,12 +75,38 @@ class FunctionRegistry extends Registry<FunctionDescription> {
     }
     const descr = addMetaInfoFromArg(addDescr);
     validateArguments(descr.args);
-    this.mapping[name] = createComputeFunctionFromDescription(descr);
+    this.mapping[name] = addInputHandling(descr, createComputeFunctionFromDescription(descr));
     super.add(name, descr);
     return this;
   }
 }
 
+function addInputHandling(
+  descr: FunctionDescription,
+  computeFunction: ComputeFunction<Arg, Matrix<ValueAndFormat> | ValueAndFormat>
+) {
+  return function (this: EvalContext, ...args: Arg[]) {
+    for (let i = 0; i < args.length; i++) {
+      const argDefinition = descr.args[descr.getArgToFocus(i + 1) - 1];
+      const acceptMatrix = argDefinition.type.some((t) => t.startsWith("RANGE"));
```

Comment:
I can measure a performance hit of ~+4.5% (on RNG's spreadsheet)
maybe we can try to move those two lines to check if the arg accepts a matrix outside of `return function (this: EvalContext, ...` to compute it only once (keep an array of booleans)

Reply:
@LucasLefevre  

to know if arg accept matrix or not, an attribute was added on `argFunctionDecription`.
The attribute is computed when formula is added to the formula registry.

If ok for you, I fixup the last commit